src/main/scala/io/github/pauljamescleary/petstore/infrastructure/repository/doobie/DoobieUserRepositoryInterpreter.scala

```diff
@@ -48,24 +47,26 @@ class DoobieUserRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
 
   import UserSQL._
 
-  def create(user: User): F[User] =
+  def put(user: User): F[User] =
     insert(user).withUniqueGeneratedKeys[Long]("ID").map(id => user.copy(id = id.some)).transact(xa)
 
-  def update(user: User): F[Option[User]] = OptionT.fromOption[ConnectionIO](user.id).semiflatMap { id =>
-    UserSQL.update(user, id).run
-  }.value.as(user.some).transact(xa)
+  def updateSafe(user: User): OptionT[F, User] = OptionT.fromOption[F](user.id).flatMapF { id =>
```

Comment:
why "updateSafe" ?

Reply:
Oops, yeah, I think this made sense in #79, but that PR got too big and should be aborted.